### ISSUE TITLE
Attempted to reset admin password, db migration multiple errors, restart.sh fails to start multiple components

### ISSUE THREAD

**Issue by gpeters86** [gchq/stroom#2894]
Running the docker implementation of stroom, 173b

Followed this guide (sort of)
https://gchq.github.io/stroom-docs/latest/docs/user-guide/tools/command-line/

I couldn't find any "stroom-app.jar", looking in either the docker containers or other stroom directories, did find "stroom-app-all.jar" and used this instead.

We can't use email to reset password because there is no email. Current password is unknown, there are no other admin users besides 'admin'.
Also unable to copy in the error logs as we can't get anything off that system due to security policy.

After running the reset password command, the password did not reset. The command line process showed numerous java errors, most relating to database migration (why does database migration happen during password reset?). Guessing this is what broke everything.

After this issue attempting to run stop.sh, start.sh; or restart.sh, results in multiple components not starting. (nginx, proxy-local, proxy-remote, stroom-log-sender) showing status : stopped.

I then attempted to run the password reset command again, and it worked successfully with no errors.

However after this success, running stop.sh/start.sh or restart.sh still results in nginx, proxy-local, proxy-remote & stroom-log-sender still show status STOPPED.

Is there any hope of fixing or has the failed database migration broken things beyond repair?

I do have a full backup I could restore from - perhaps if anyone could provide guidance on what parts might have broken during the migration failure, I could do just a partial restore. (I'm not sure if the full backup restore would actually work, as we've only just setup the backups (luckily) and are yet to test a restore.

Also what would the correct command line password reset process be? I know I did it once a long time ago, but can't recall if I found stroom-app.jar, or did it from somewhere else like in the database manually (if that's where the password is even stored?)

**Comment by stroomdev10** [comment]
The reset procedure assumes that you have a working and running Stroom.  If you are using a stack, you should exec into the running stroom container to run the reset_password command from within stroom directory.  stroom-app-all.jar is the correct jar , you must also specify the config.yml (found in the config folder).

Assuming you did this correctly I can't see why Stroom would be trying to perform migrations. If a *later* version of stroom-app-all.jar is used, then it will attempt to run any outstanding upgrades as per https://gchq.github.io/stroom-docs/latest/docs/user-guide/tools/command-line/#authentication-accounts-and-stroom-users  

...
This command will also run any necessary database migrations to ensure it is working with the correct version of the database schema.
...

Not sure why nginx and stroom-log-sender are STOPPED tbh, they don't have a dependency on stroom.  Do the docker logs provide any clues as to why they are stopped.

Are the DB and Stroom containers up and working OK?

**Comment by at055612** [comment]
I will add a new `command.sh` script to the stack that will enable you to run any of the commands, e.g.
`./command.sh reset_password -u foo -p bar`

I terms of your broken stack, please can you explain exactly how you ran the reset_password command, e.g. exactly what command you ran and where from (e.g. inside the container or outside).  Where did you find the stroom-app-all.jar that you used?

>why does database migration happen during password reset?
Every time stroom starts it verifies the state of the database to ensure all migrations it expects to be have been run.  This is the same for the commands like reset_password.  If the DB is up to date which it should have been the no migrations will happen.
This suggests that it was either pointing at a different DB or you were running reset_password using a jar file for a later version of stroom than you have in your stack.

**Comment by gpeters86** [comment]
Thanks again!
The stroom-app-all.jar was found inside the containers, in directories such as /stroom, /stroom-proxy, etc.

It seems like everything works now, The database rebuild required more space than was available, when I ran restart/start and it  tried to rebuild. (I guess, still figuring Stroom's finer details out).
(Unfortunately the physical server is lacking, there's not much more I can do about it except slowly learning where the logs are growing and setup tails/ tighten rotation as I discover new offending log locations in the containers)

So the password reset actually worked, everything else just broke when the migration was attempted with insufficient space I'm guessing.

**Comment by at055612** [comment]
Glad to hear it is working now.  Are you happy to close this now?

**Comment by gpeters86** [comment]
> Glad to hear it is working now. Are you happy to close this now?

Yes, go ahead. thanks. - Just a quick addition before if you don't mind. stroom-app-all.jar seems to work, but where should I actually find stroom-app.jar? the documentation says "in the root directory" but I'm not sure what that means, given there are multiple containers, and I can't find it in the base stroom folder where start.sh, the config directory, etc are found.
